### ISSUE TITLE
Custom resolvers doens't  overrides and attached to respective mutation/query

### ISSUE THREAD

**Issue by hackrx** [aws-amplify/amplify-cli#9348]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### Amplify CLI Version

7.6.3

### What operating system are you using?

Mac

### Amplify Categories

auth, storage, function, api

### Amplify Commands

push

### Describe the bug

As described [here](https://docs.amplify.aws/cli/graphql/custom-business-logic/#override-amplify-generated-resolvers), I followed the guide to override `createUser` req and req  vtl which is automatically generated by appsync.
## schema.graphql
```
.
.
.
.

input CreateUserInput {
  id: ID
   name: String
  emailID: String
  mobileNumber: String
}

type mutation{
createUser(input: CreateUserInput): User @aws_iam @aws_cognito_user_pools
}

....
```
## Mutation.createUser.req.vtl
```
## [Start] Create Request template. **
## Set the default values to put request **
#set( $mergedValues = $util.defaultIfNull($ctx.stash.defaultValues, {}) )
## copy the values from input **
$util.qr($mergedValues.putAll($util.defaultIfNull($ctx.args.input, {})))
## set the typename **
$util.qr($mergedValues.put("__typename", "User"))
#set( $PutObject = {
"version": "2018-05-29",
  "operation": "PutItem",
  "attributeValues":   $util.dynamodb.toMapValues($mergedValues),
  "condition": $condition
} )
#if( $context.args.condition )
  $util.qr($ctx.stash.conditions.add($context.args.condition))
#end
## Begin - key condition **
#if( $ctx.stash.metadata.modelObjectKey )
  #set( $keyConditionExpr = {} )
  #set( $keyConditionExprNames = {} )
  #foreach( $entry in $ctx.stash.metadata.modelObjectKey.entrySet() )
    $util.qr($keyConditionExpr.put("keyCondition$velocityCount", {
  "attributeExists": false
}))
    $util.qr($keyConditionExprNames.put("#keyCondition$velocityCount", "$entry.key"))
  #end
  $util.qr($ctx.stash.conditions.add($keyConditionExpr))
#else
  $util.qr($ctx.stash.conditions.add({
  "id": {
      "attributeExists": false
  }
}))
#end
## End - key condition **
## Start condition block **
#if( $ctx.stash.conditions && $ctx.stash.conditions.size() != 0 )
  #set( $mergedConditions = {
  "and": $ctx.stash.conditions
} )
  #set( $Conditions = $util.parseJson($util.transform.toDynamoDBConditionExpression($mergedConditions)) )
  #if( $Conditions.expressionValues && $Conditions.expressionValues.size() == 0 )
    #set( $Conditions = {
  "expression": $Conditions.expression,
  "expressionNames": $Conditions.expressionNames
} )
  #end
  ## End condition block **
#end
#if( $Conditions )
  #if( $keyConditionExprNames )
    $util.qr($Conditions.expressionNames.putAll($keyConditionExprNames))
  #end
  $util.qr($PutObject.put("condition", $Conditions))
#end
#if( $ctx.stash.metadata.modelObjectKey )
  $util.qr($PutObject.put("key", $ctx.stash.metadata.modelObjectKey))
#else
  #set( $Key = {
  "id":   $util.dynamodb.toDynamoDB($mergedValues.id)
} )
  $util.qr($PutObject.put("key", $Key))
#end
$util.toJson($PutObject)
## [End] Create Request template. **
```

## Mutation.createUser.res.vtl
```
## [Start] ResponseTemplate. **
$util.qr($ctx.result.put("__operation", "Mutation"))
#if( $ctx.error )
    $util.error($ctx.error.message, $ctx.error.type)
#else
    $util.toJson($ctx.result)
#end
## [End] ResponseTemplate. **

```

And in the appsync console, there is nothing attached, I am getting null on executing `createUser` mutation.

![image](https://user-images.githubusercontent.com/54453778/146785621-a437b01a-60cc-4a7c-a7fa-7fd178cd2fa3.png)

### Expected behavior

it should attach the resolver.

### Reproduction steps

1. create a `User` modal
2. add `mutations:{create:null}` on the modal
3. Add `createUser` in `type mutation`
4. create req,res resolver file `resolvers/` folder
5. amplify push
6. now check the console.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by cjihrig** [comment]
I cannot reproduce this based on the information provided. I created the following schema:

```graphql
type User @model(mutations: {create: null}) {
  name: String
  emailID: String
  mobileNumber: String
}
```

Note that you have disabled the create operation with `mutations: {create: null}`.

I also added a custom `resolvers/Mutation.createUser.req.vtl` file. After running `amplify api gql-compile`, I am seeing the same content in `build/resolvers/Mutation.createUser.req.vtl`.

**Comment by aryaman-titan** [comment]
Hey @cjihrig! I was facing similar issue. Even after disabling the create operation, createUser is getting populated in the `/build` folder

**Comment by cjihrig** [comment]
> Even after disabling the create operation, createUser is getting populated in the /build folder

If you have defined custom VTL files in the `resolvers/` directory, they will be copied to the `build/resolvers/` directory, but the generated schema should not include the functionality for invoking them from the deployed AppSync API.

**Comment by hackrx** [comment]
Hey @cjihrig, I noticed on thing, if I will mark `{mutation: Create}` as null, then it will not autogenerate any mutation for it, also I have the same file inside `/resolver/` (not inside build), it is getting created in build folder on `gql-compile` but if you push it, then it will not populate in appsync console, weird issue, it must auto-connect with createUser mutation, and take User as datasource.
So to fix this issue, I removed `mutations: {create: null}` from the model, and then pasted the same file(`Mutation.createUser.req.vtl` updated vtl) in resolver folder. And it worked. Also, I was able to override the `CreateUserInput` by simply writing the same inside `schema.graphql`. 
Thanks for the follow-up.
